src/jit/lclvars.cpp

```diff
@@ -4044,12 +4086,14 @@ int Compiler::lvaAssignVirtualFrameOffsetToArg(unsigned lclNum, unsigned argSize
 #if defined(_TARGET_X86_)
         argOffs += sizeof(void *);
 #elif defined(_TARGET_AMD64_)
-#ifndef UNIX_AMD64_ABI
-        // The offset for args need not to be set for System V.
-        // No outgoing args for callees. The spilled args offsets will be set later.
-        varDsc->lvStkOffs = argOffs;
-        argOffs += sizeof(void *);
-#endif // !UNIX_AMD64_ABI
+#ifdef UNIX_AMD64_ABI
+        if (varDsc->lvOnFrame)
+#endif
+        {
+            // The offset for args need to be set only for the stack homed arguments for System V.
+            varDsc->lvStkOffs = argOffs;
+            argOffs += sizeof(void *);
+        }
 #elif defined(_TARGET_ARM64_)
         // Register arguments don't take stack space.
```

Comment:
Also you might have to implement logic here for promoted struct as this is no longer true.